proxy/ha_proxy.go

```diff
@@ -174,26 +174,32 @@ func (m HaProxy) getConfigData() ConfigData {
 	d := ConfigData{
 		CertsString: strings.Join(certs, " "),
 	}
-	d.ConnectionMode = m.getSecretOrEnvVar("CONNECTION_MODE", "http-server-close")
-	d.TimeoutConnect = m.getSecretOrEnvVar("TIMEOUT_CONNECT", "5")
-	d.TimeoutClient = m.getSecretOrEnvVar("TIMEOUT_CLIENT", "20")
-	d.TimeoutServer = m.getSecretOrEnvVar("TIMEOUT_SERVER", "20")
-	d.TimeoutQueue = m.getSecretOrEnvVar("TIMEOUT_QUEUE", "30")
-	d.TimeoutTunnel = m.getSecretOrEnvVar("TIMEOUT_TUNNEL", "3600")
-	d.TimeoutHttpRequest = m.getSecretOrEnvVar("TIMEOUT_HTTP_REQUEST", "5")
-	d.TimeoutHttpKeepAlive = m.getSecretOrEnvVar("TIMEOUT_HTTP_KEEP_ALIVE", "15")
-	d.StatsUser = m.getSecretOrEnvVar("STATS_USER", "admin")
-	d.StatsPass = m.getSecretOrEnvVar("STATS_PASS", "admin")
-	usersString := m.getSecretOrEnvVar("USERS", "")
+	d.ConnectionMode = GetSecretOrEnvVar("CONNECTION_MODE", "http-server-close")
+	d.TimeoutConnect = GetSecretOrEnvVar("TIMEOUT_CONNECT", "5")
+	d.TimeoutClient = GetSecretOrEnvVar("TIMEOUT_CLIENT", "20")
+	d.TimeoutServer = GetSecretOrEnvVar("TIMEOUT_SERVER", "20")
+	d.TimeoutQueue = GetSecretOrEnvVar("TIMEOUT_QUEUE", "30")
+	d.TimeoutTunnel = GetSecretOrEnvVar("TIMEOUT_TUNNEL", "3600")
+	d.TimeoutHttpRequest = GetSecretOrEnvVar("TIMEOUT_HTTP_REQUEST", "5")
+	d.TimeoutHttpKeepAlive = GetSecretOrEnvVar("TIMEOUT_HTTP_KEEP_ALIVE", "15")
+	d.StatsUser = GetSecretOrEnvVar("STATS_USER", "admin")
+	d.StatsPass = GetSecretOrEnvVar("STATS_PASS", "admin")
+	usersString := GetSecretOrEnvVar("USERS", "")
+	encryptedString := GetSecretOrEnvVar("USERS_PASS_ENCRYPTED", "")
 	if len(usersString) > 0 {
 		d.UserList = "\nuserlist defaultUsers\n"
+		passwordType := "insecure-password";
+		if encryptedString == "true" || encryptedString == "1" {
+			passwordType = "password";
+		}
 		users := strings.Split(usersString, ",")
 		for _, user := range users {
-			userPass := strings.Split(user, ":")
-			d.UserList = fmt.Sprintf("%s    user %s insecure-password %s\n", d.UserList, userPass[0], userPass[1])
+			//trimming to allow new lines in file
+			userPass := strings.Split(strings.Trim(user, "\n\t "), ":")
```

Comment:
disagree GetSecretOrEnvVar trimms whole content where this trims a line so 
```
         asasdasd:Asdasdasd       ,
  asdasd:asdasdasd,
asfdasfdsf:asdfsdfsdfffdfd
```
wont include spaces in user names/passwords.